agent/engine/data.go

```diff
@@ -68,6 +68,21 @@ func (engine *DockerTaskEngine) removeTaskData(task *apitask.Task) {
 	}
 }
 
+func (engine *DockerTaskEngine) removeENIAttachment(mac string) {
+	attachmentToRemove, ok := engine.state.ENIByMac(mac)
+	if ok {
```

Comment:
Should it throw an error when it's not ok

Reply:
Done.

Reply:
`if ok` not required here now